pennylane/gradients/vjp.py

```diff
@@ -136,7 +128,12 @@ def compute_vjp_single(dy, jac, num=None):
         if num == 1:
             jac = qml.math.squeeze(jac)
         jac = qml.math.reshape(jac, (-1, 1))
-        res = qml.math.tensordot(jac, dy_row, [[0], [0]])
+        try:
+            res = dy_row @ jac
+
+        except Exception:  # pylint: disable=broad-exception-caught
+            res = qml.math.tensordot(jac, dy_row, [[0], [0]])

```

Comment:
So here we are trying the fast path and falling back to the slower tensordot if anything goes wrong?

Reply:
Yes. This gave us seconds fewer of runtime in situations where the first path was able to complete. We can improve on this later, but this for now was the cheapest way to get a more improved runtime within the slow-path.